interpreter/CMakeLists.txt

```diff
@@ -9,6 +9,22 @@ set(LLVM_BUILD_TOOLS OFF CACHE BOOL "")
 set(LLVM_TOOL_LLVM_AR_BUILD OFF CACHE BOOL "")
 set(CLANG_TOOL_CLANG_OFFLOAD_BUNDLER_BUILD OFF CACHE BOOL "")
 set(LLVM_FORCE_USE_OLD_TOOLCHAIN ON CACHE BOOL "")
+
+# FIXME: We cannot use etcdir or prefix from RootConfiguration.cmake because it
+# is included very late.
+if(gnuinstall)
+  set(prefix ${CMAKE_INSTALL_PREFIX})
+else()
+  set(prefix ${ROOTSYS})
+endif()
+if(IS_ABSOLUTE ${CMAKE_INSTALL_SYSCONFDIR})
+  set(etcdir ${CMAKE_INSTALL_SYSCONFDIR})
+else()
+  set(etcdir ${prefix}/${CMAKE_INSTALL_SYSCONFDIR})
+endif()
+set(CLING_PLUGIN_INSTALL_PREFIX "${etcdir}/cling/plugins/" CACHE STRING "" FORCE)
```

Comment:
`<prefix>/etc` is the wrong place for plugins. Only small text configuration files should go into `<prefix>/etc`.
Better choices would be `<prefix>/lib/cling` or `<prefix>/share/cling/plugins`. ROOT would make my `/etc` about 10 times bigger, for instance:
```
epsft-53 ~ # du -shc /etc
15M	/etc
15M	total
epsft-53 ~ # du -shc /opt/root/6.12/etc
116M	/opt/root/6.12/etc
116M	total
```

Reply:
IIUC, our ROOT-based plugins live there, too.

Reply:
True, but they will also need to be moved to a more appropriate place. The situation above is not acceptable. We need to start minimally following the FHS from now on, to be able to reach the point in which we will become a regular package in most Linux distributions. This is another point for “let's not make it worse if we can avoid it”.

Reply:
Ok, I'd pick `<prefix>/share/cling/plugins`.

Reply:
Hi Vassil, thanks for understanding. How many such plugins exist now, and are users expected to install more plugins separately? Actually, I have at least these on my machine:
```
/usr/lib64/krb5/plugins
/usr/lib64/qt4/plugins
/usr/lib64/nsbrowser/plugins
/usr/lib64/libv4l/plugins
/usr/lib64/qt5/plugins
/usr/lib64/bluetooth/plugins
/usr/lib64/peas-demo/plugins
/usr/lib64/eog/plugins
/usr/lib64/traceevent/plugins
```
Looking at [FHS](http://refspecs.linuxfoundation.org/FHS_3.0/fhs/index.html) to check, `<prefix>/share` is meant for architecture-independent data files, so I'm inclined to use `<prefix>/lib/root/plugins` for ROOT, and suggest that we use `<prefix>/lib/cling/plugins` for cling for consistency, what do you think? The rationale is that these are executables, so probably `<prefix>/lib` makes more sense.

Reply:
Hi Guilherme,

I understand and agree with the principle of 'not making things harder for an upcoming change'.  In this case, however, doesn't putting Clad's file in a 'different' (but better) place than the others more confusing than helpful.  (i.e. it will help the developer later but will confused the users now).   What do you think and/or what am I missing?

Cheers,
Philippe.

Reply:
Hi Philippe, I disagree that installing files where FHS indicates is confusing. There is a problem that /etc is hard-coded in a few places for the other files now, so if this goes to the right place since the beginning it will be easier to fix the rest later, and it will avoid the appearance of more hard-coded paths where it only works with the default value. For example, changing SYSCONFDIR breaks ROOT right now, that's why I couldn't change it yet.

Reply:
> I disagree that installing files where FHS indicates is confusing. 

You might be misunderstanding my point.  I agree with you that "where FHS indicates" is the better place to install the files and what we should strive for.  My concern is the in-between situation where/when we have some files installed "where FHS indicates" and some in other places.  (i.e semantically similar things installed in two different places) ... if we **already** have the case that some files are installed "where FHS indicates" then my point is  moot (because the confusing situation I fear is already there).